src/rpdk/core/contract/resource_generator.py

```diff
@@ -32,12 +33,12 @@
 # date is extraction from date-time
 # time is extraction from date-time
 STRING_FORMATS = {
-    "arn": "^arn:aws(-(cn|gov))?:[a-z-]+:(([a-z]+-)+[0-9])?:([0-9]{12})?:[^.]+$",
+    "arn": "^arn:aws(-(cn|us-gov))?:[a-z-]+:([a-z]{2}-([a-z]+-)+[0-9])+:([0-9]{12}){1}:[a-z0-9-:/*]+$",
     "uri": "^(https?|ftp|file)://[0-9a-zA-Z]([-.\\w]*[0-9a-zA-Z])(:[0-9]*)*([?/#].*)?$",
     "date-time": r"^(\d{4})-(0[1-9]|1[0-2])-(\d{2})T(?:[01]\d|2[0123]):(?:[0-5]\d):(?:[0-5]\d)(?:\.\d+)?(?:Z|[+-](?:[01]\d|2[0123]):[0-5]\d)$",
     "date": r"^(\d{4})-(0[1-9]|1[0-2])-(\d{2})$",
     "time": r"^(?:[01]\d|2[0123]):(?:[0-5]\d):(?:[0-5]\d)(?:\.\d+)?(?:Z|[+-](?:[01]\d|2[0123]):[0-5]\d)$",
-    "email": r"^.+@[^\.].*\.[a-z]{2,}$",
+    "email": r"^[a-zA-Z0-9+_-]+@[a-zA-Z0-9_-]+\.[a-zA-Z]{2,}$",
```

Comment:
I think there are a lot more characters that make an email address valid.  What is wrong with the current regex?

Reply:
We can revert to the previous email regex.  Was attempting to reduce some invalid characters.